pkg/kubelet/containerdshim/containerd_container.go

```diff
@@ -162,38 +173,54 @@ func (cs *containerdService) CreateContainer(podSandboxID string, containerConfi
 
 	// mikebrow TODO proper console handling
 	glog.V(2).Infof("CreateContainer for container %s container directory %s", containerID, containerDir)
-	response, err := cs.cdClient.Create(gocontext.Background(), create)
+	response, err := cs.containerService.Create(gocontext.Background(), create)
 	if err != nil {
 		return "", err
 	}
 
+	containerStoreLock.Lock()
```

Comment:
Good point. Actually I also thought about this. When `Create` returns error:
1) If we add state before `Create`, the container will be in `Exited` state, because there is no corresponding containerd container;
2) If we add state after `Create`, the container will not be in the list.

To me, either is ok, as long as containerd cleans up properly after Create failure.